### PR TITLE
Feature sunlight coverage

### PR DESCRIPTION
**Pull request by mraspaud**
This PR adds a plugin to check for sunlight coverage over areas.

### CODE REVIEW

File: trollflow2/__init__.py [hunk 1]
```diff
@@ -314,6 +314,53 @@ def sza_check(job):
...
+    from pyresample.boundary import AreaDefBoundary
+    from trollsched.spherical import get_twilight_poly
+
+    adp = AreaDefBoundary(area_def, frequency=100)
+    poly = get_twilight_poly(start_time)
+
+    daylight = adp.contour_poly.intersection(poly)
+    if daylight is None:
+        if sun_zenith_angle(start_time, *area_def.get_lonlat(0, 0)) < 90:
+            return 1
```
Comment (pnuu) [inline trollflow2/__init__.py]: Should the return values be floats?

File: trollflow2/__init__.py [hunk 2]
```diff
@@ -314,6 +314,53 @@ def sza_check(job):
...
+    """Get the sunlight coverage of *area_def* at *start_time*."""
+    from pyresample.boundary import AreaDefBoundary
+    from trollsched.spherical import get_twilight_poly
+
+    adp = AreaDefBoundary(area_def, frequency=100)
+    poly = get_twilight_poly(start_time)
+
+    daylight = adp.contour_poly.intersection(poly)
+    if daylight is None:
+        if sun_zenith_angle(start_time, *area_def.get_lonlat(0, 0)) < 90:
```
Comment (pnuu) [inline trollflow2/__init__.py]: Is check in one corner of the area enough?

File: trollflow2/__init__.py [hunk 3]
```diff
@@ -314,6 +314,53 @@ def sza_check(job):
             dpath.util.delete(product_list, '/product_list/%s' % area)
 
 
+def check_sunlight_coverage(job):
+    """Remove products with too low daytime coverage.
+
+    This plugins looks for a parameter called `min_sunlight_coverage` in the
+    product list, expressed in % (so between 0 and 100). Everything under it not
+    matching this sunlight coverage will be discarded.
```
Comment (pnuu) [inline trollflow2/__init__.py]: The last sentence needs some rewording to be more understandable. Maybe something like `If the sunlit fraction is less than configured, the affected products will be discarder.`?

File: setup.py [hunk 4]
```diff
@@ -51,7 +51,7 @@
       data_files=[],
       zip_safe=False,
       install_requires=install_requires,
-      tests_require=['mock'],
+      tests_require=['mock', 'pytroll-schedule', 'pyresample'],
```
Comment (pnuu) [inline setup.py]: Do we need these packages for the tests or could (should?) they be mocked?

File: trollflow2/__init__.py (hunk 1 above)
Comment (mraspaud) [reply trollflow2/__init__.py]: I can do that

File: trollflow2/__init__.py (hunk 2 above)
Comment (mraspaud) [reply trollflow2/__init__.py]: in principle yes as, having an empty intersection means either full light, or full night

File: setup.py (hunk 4 above)
Comment (mraspaud) [reply setup.py]: I can mock them if you prefer

File: trollflow2/__init__.py (hunk 3 above)
Comment (mraspaud) [reply trollflow2/__init__.py]: ok

**Pull request merged**